Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs

```diff
@@ -0,0 +1,242 @@
+﻿using NServiceBus;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Configuration;
+using System.Web.Configuration;
+using System.Web.Hosting;
+
+class AzureConfigurationSourceReplacement
+{
+    AzureConfigurationSourceReplacement(EndpointConfiguration endpointConfiguration)
+    {
+        #region azure-configuration-source-replacement
+
+        var sectionName = "mySection";
+        var attributeName = "myAttribute";
+        string value;
+        if (SafeRoleEnvironment.IsAvailable)
+        {
+            var key = sectionName + "." + attributeName;
+            value = SafeRoleEnvironment.GetConfigurationSettingValue(key);
+        }
+        else
+        {
+            var section = ConfigurationResolver.GetConfigurationHandler()
+                              .GetSection(sectionName) as MyConfigurationSection;
+            value = section.MyAttribute;
+        }
+
+        #endregion
+    }
+}
+
+#region configuration-resolver
+
+static class ConfigurationResolver
+{
+    public static Configuration GetConfigurationHandler()
+    {
+        if (IsWebsite()) return WebConfigurationManager.OpenWebConfiguration(HostingEnvironment.ApplicationVirtualPath);
```

Comment:
Condition on the same line with execution? Let's do it right, on separate lines. 🙂